File: WorldFeed/Server/src/WorldFeed.Common/Public/Events/ExceptionEventArgs.cs
namespace WorldFeed.Common.Public.Events
{
    using System;

    using WorldFeed.Common.Public.Streaming.Events;

    /// <summary>
    /// Event to inform that a stream has stopped.
    /// </summary>
    public class StreamStoppedEventArgs : EventArgs
    {
        public StreamStoppedEventArgs()
        {
        }

        public StreamStoppedEventArgs(Exception ex, IDisconnectMessage disconnectMessage = null)
        {
            Exception = ex;
            DisconnectMessage = disconnectMessage;
        }

        public Exception Exception { get; }

        public IDisconnectMessage DisconnectMessage { get; }
    }
}

File: Sharebook/Client/src/app/core/Public/Events/ExceptionEventArgs.ts
import {IDisconnectMessage} from '../Streaming/Events/IDisconnectMessage';
import Exception from 'typescript-dotnet-commonjs/System/Exception';

// Event to inform that a stream has stopped.
export class StreamStoppedEventArgs /*extends EventArgs*/ {
  constructor(ex?: Exception, disconnectMessage: IDisconnectMessage = null) {
    // super();
    if (ex) {
      this.Exception = ex;
      this.DisconnectMessage = disconnectMessage;
    }
  }

  public Exception: Exception;
  public DisconnectMessage: IDisconnectMessage;
}

// public StreamStoppedEventArgs()
// {
// }
//
// public StreamStoppedEventArgs(Exception ex, IDisconnectMessage disconnectMessage = null)
// {
//     Exception = ex;
//     DisconnectMessage = disconnectMessage;
// }
